deblur/workflow.py

```diff
@@ -207,6 +208,101 @@ def build_index_sortmerna(ref_fp, working_dir):
     return all_db
 
 
+def filter_minreads_samples_from_table(table, minreads=1, inplace=True):
+    """Filter samples from biom table that have less than
+    minreads reads total
+
+    Paraneters
+    ----------
+    table : biom.Table
+        the biom table to filter
+    minreads : int (optional)
+        the minimal number of reads in a sample in order to keep it
+    inplace : bool (optional)
+        if True, filter the biom table in place, if false create a new copy
+
+    Returns
+    -------
+    table : biom.Table
+        the filtered biom table
+    """
+    logger = logging.getLogger(__name__)
+    logger.debug('filter_minreads_started. minreads=%d' % minreads)
+    samp_sum = table.sum(axis='sample')
+    samp_ids = table.ids(axis='sample')
+    bad_samples = samp_ids[samp_sum < minreads]
+    if len(bad_samples) > 0:
+        logger.warn('removed %d samples with reads per sample<%d'
+                    % (len(bad_samples), minreads))
+        table = table.filter(bad_samples, axis='sample',
+                             inplace=inplace, invert=True)
+    else:
+        logger.debug('all samples contain > %d reads' % minreads)
+    return table
+
+
+def remove_artifacts_from_biom_table(table_filename,
+                                     fasta_filename,
+                                     ref_fp,
+                                     biom_table_dir,
+                                     ref_db_fp,
+                                     threads=1,
+                                     verbose=False,
+                                     sim_thresh=None,
+                                     coverage_thresh=None):
+    """Remove artifacts from a biom table using SortMeRNA
+
+    Parameters
+    ----------
+    table : str
+        name of the biom table file
+    fasta_filename : str
+        the fasta file containing all the sequences of the biom table
+    """
+    logger = logging.getLogger(__name__)
+    logger.info('getting 16s sequences from the biom table')
+
+    # remove artifacts from the fasta file. output is in clean_fp fasta file
+    clean_fp = remove_artifacts_seqs(fasta_filename, ref_fp,
+                                     working_dir=biom_table_dir,
+                                     ref_db_fp=ref_db_fp,
+                                     negate=False, threads=threads,
+                                     verbose=verbose,
+                                     sim_thresh=sim_thresh,
+                                     coverage_thresh=coverage_thresh)
+    logger.debug('removed artifacts from sequences input %s'
+                 ' to output %s' % (fasta_filename, clean_fp))
+
+    # read the clean fasta file
+    good_seqs = set()
+    for chead, cseq in sequence_generator(clean_fp):
```

Comment:
The only "technical" issue, which we could also ignore, with @wasade's version is that if clean_fp is empty it will create an empty dict, not an empty set. As you can imagine this is completely out of the scope of this PR 😃 .